### DISCUSSION TITLE
Provide SMTP access

### DISCUSSION THREAD

**Discussion by EthraZa** [simple-login/app#679] [Feature Requests]
This is a request for a much needed **SMTP access** feature to take [Simplelogin.co](https://Simplelogin.co) to an _all new enterprise level_.

Right now, if we migrate our company emails to SL, for example, if we need to send a single email to a random costumer or partner and have to go all the way to create a reverse-alias, that would be a pain and basically prevent us to adopt SL as our enterprise email solution.

If SL opens up SMTP so we can login with aliases, even with a heavy limit on sending amount by hour/day and even in a new paying tier, that would be superb.
SL already allow us to send e-mails, so it would be nothing really new, it would just send e-mail using our aliases instead of using our reverse-aliases - Simple like that.

>  For all of you who would like to send daily email with your alias, without resorting to the reverse-alias feature, please, 
> - [X] vote it up.

Then It would be possible to do something like this:

ie. on Gmail, on Config, Send E-mail as...

```yml
SMTP host: smtp.simplelogin.co
Port: 587
User: alias@mydomain.com
Password: "An API Key"
```

- Edited to be a feature request

**Comment by nguyenkims** [comment]
@EthraZa SimpleLogin doesn't provide a direct SMTP access and currently, the only way of sending emails is to use reverse alias. As SimpleLogin isn't a transactional email service (e.g. SendGrid), we don't open SMTP access.

**Reply by EthraZa** [reply]
With a service as good as you have, I hope SL can expand into transactional email service in the future. :)

**Reply by nguyenkims** [reply]
Thanks :) but transactional email service is actually quite different so we don't have any plan to become one at the moment.

**Reply by EthraZa** [reply]
I was thinking about that and, at least to my intentions, I was just thinking in normal daily use, not bulk sending.

I was thinking in take my company email to SL but that would be a pain when we need to send a single email to a random costumer or partner and have to go all the way to create a reverse-alias.
If you guys could consider to open SMTPS (to login with aliases), even with a heavy limit on sending amount by hour/day and even in a new paying tier, that would be superb.

Just think about. I would be your first costumer in this new paying tier. ;)

**Reply by trikotret** [reply]
I tried to us SL with my business email but like you said, it was a pain to have to make a reverse alias for each person you going to email. The  other day i accidently sent an email to a client with 2 recipients, one got my business email and the other got my personal Gmail. I guess I forgot to make a reverse alias for the other person. lol

**Reply by nguyenkims** [reply]
Oh I see, I have converted this thread to a feature request so other people can upvote on it then. And feel free to upvote it to make it happen sooner :).

**Reply by EthraZa** [reply]
Nice

**Reply by ghost** [reply]
i hope you will make this opt-in if you do it (or protect it by at least some 2FA mechanism) like for instance concat APIkey and OTP" in the api key (first time only) like that your server check the api key if valid check the otp first time to identify and after that totally ignore the otp part of the api key.

**Reply by xenstar** [reply]
> @EthraZa SimpleLogin doesn't provide a direct SMTP access and currently, the only way of sending emails is to use reverse alias. As SimpleLogin isn't a transactional email service (e.g. SendGrid), we don't open SMTP access.

Just like Esthraza said, we just need it for daily use. Maybe you can limit the sending to 20-50 emails per hour, in this way, people will not abuse the service. But SMTP feature will take simple login to another level for sure.

**Reply by trikotret** [reply]
I doubt they'll add it since simpelogin joined Protonmail.

**Reply by xenstar** [reply]
> I doubt they'll add it since simpelogin joined Protonmail.

I contacted their support, they said they already have it on the timeline and they will launch it after a few months. It seems like they are just afraid, that people will abuse their SMTP server.

**Reply by edsimpsons83** [reply]
> > I doubt they'll add it since simpelogin joined Protonmail.
> 
> I contacted their support, they said they already have it on the timeline and they will launch it after a few months. It seems like they are just afraid, that people will abuse their SMTP server.

Protonmail provides SMTP access, you think they could just duplicate the quota / throttling that they use.

**Comment by trikotret** [comment]
This would be sick if you can add smtp access to SL. You definitely would not need google workspace anymore

**Reply by xenstar** [reply]
You are right.

**Comment by mygrexit** [comment]
That would be so awesome. Especially for people with websites/forums.

**Reply by armandolas** [reply]
That would be great, ImprovMX has this feature, which allows the use of "send mail as" in Gmail instead of having to create a reverse-alias to every new email we want to write

**Comment by trikotret** [comment]
This is the perfect time to get it done since google is dropping the free legacy Gsuite accounts on July. Im sure lots of gsuite users are looking for alternatives to take their domains.

**Reply by ghost** [reply]
Me for that i use a paid tier of tutanota (because twelve buks to support them isn't really much, but nothing prevent you to do the same on free tiers), and reverse alias to send message from my mail.

You can do it too with proton if you prefer (i have choosen tutanota for their "in dev" quantum resistent encryption algo and because it's cheaper than protonmail overpriced).

**Reply by twratl** [reply]
I came here looking for this! I agree it is an opportunity to steal some business from Google. I am most certainly looking for alternatives and would love to see SimpleLogin build out this feature.

**Reply by ghost** [reply]
> I came here looking for this! I agree it is an opportunity to steal some business from Google. I am most certainly looking for alternatives and would love to see SimpleLogin build out this feature.

The problem it's the feature request hitself to replace google Simple login must pass from "alias services" to "email provider".
So instead of "provide SMTP access" you must ask "Provide Email services".

**Reply by edsimpsons83** [reply]
Agreed, with Google getting rid of the free legacy Gsuite, I am looking to move and this feature would be awesome.

**Comment by developStorm** [comment]
+1 for this. Implementing SMTP will also allow better forwarding. It will not be necessary to modify any of the original mail headers, remove need for changing sender address and keep the original DKIM signature valid, as the replacement can be done by inbond SMTP server transparently.

**Comment by jbarr** [comment]
Obviously this would change the scope of SL, but it would make SL a one stop shop, so please take my money. BUT I do get it that you should also focus on doing one or a few things extremely well instead of trying to be ineffectively all things. Still I'm up voting.

**Comment by sahilph** [comment]
Hello all,

I have created a PR #869  for this. Please check.

Thanks.

**Reply by xenstar** [reply]
Thank you so much. It looks promising, hope the team will add this feature soon.

**Comment by EvilAaron** [comment]
edit: just reading PR #869 , looks exciting, can't wait for it to roll out will solve that 1 fly ;)

Just setting up SimpleLogin here for the first time, imported some domain names and really like it and likely buying a year plan.

The fly in the ointment is no SMTP for "reply as" support. reverse-alias is very clunky and in no way can ever pass WAF factor here. NEED a better solution :/

**Comment by kwanm84** [comment]
Just like to add my support for this feature.

I'd like to be able to have my wife use my custom domain without having to add new reverse alias every time she wants to email someone. Happy to pay (a little) extra for SMTP access.

**Comment by argarner** [comment]
Considering the Proton acquisition of SimpleLogin, how mutually beneficial this would be to both services value proposition, and the significant value-add to all domain owners, it would make a lot of sense to enable the SMTP capability for all Proton premium/paid accounts **at no additional cost**, while charging an additional fee for standalone SimpleLogin SMTP (which I imagine could still leverage a lot of the same Proton SMTP infra/code).

This would fill the gap for consumers and businesses who want to move to a privacy oriented email provider, but don't want to be forced to use Protons extremely limited suite of client applications and capabilities.

I would strongly urge both Proton and SimpleLogin to look into designing a next-generation SMTP component that encapsulates the features of both ProtonMail & SimpleLogin to supercharge the privacy and security of all users, while expanding on their existing featuresets – the capability for consumers and businesses to own their private encryption keys and data, allow the use of any client-side application, etc. The fact that Proton requires hosting of private keys has been the main criticism and limitation of Proton since it's inception (a truly private, future proof, service provider would offer the capability to function solely as a encrypted data store/relay, instead of the E2E owner of all that data stores integrations). The fact that SimpleLogin does not have zero-access encryption, and is susceptible to breaches exposing user data (including which emails are associated with every single alias) is the single largest risk in using email aliases (A single breach could destroy all of the efforts of SimpleLogin's users, in trying to obfuscate their identities from big tech and surveillance capitalism, as well as the entire value proposition of the service).

**Comment by adriy-be** [comment]
Hope this feature come soon 😁

**Comment by leoleoasd** [comment]
Hi all, for anyone who needs it, I wrote an SMTP relay for simple login:
https://github.com/leoleoasd/simple-login-smtp-relay/

It works like a regular SMTP server: you send an email to it, it reads the alias from the email and automatically creates reverse alias for all your recipients, then forward it to your actual mailbox (such as Gmail).
And I don't think that this brings people the ability to abuse it. People can already programmingly create reverse alias and send emails with their own mailbox in bulk. If we are going to avoid being abused, I think we need to limit the API calls to create contacts and a rate limit for replying to emails.

**Comment by trikotret** [comment]
I am actually using SMTP2GO Smtp access on Gmail. I setup a SMTP email on Gmail. If I need to email anyone, Ill start the email with SMTP email. When someone replies,I will then get a reverse email to reply. I  then use my main default address to reply and so on. I set up always reply from default address. SMTP2GO is free for 1000 emails a month.

**Comment by hax4dazy** [comment]
@simple-login any chance this can be looked at again? I'd assume that the PR at https://github.com/simple-login/app/pull/869 has to be reworked a bit because its heavily out of date but I am actually interested in this feature

**Comment by numanzahid** [comment]
Hope to see this feature implemented soon.

**Comment by 621625** [comment]
I also support SMTP for SimpleLogin.
